lil-lab/recnet
# [recnet-api] Add Get current user api

## Description

1. Refactor authentication decorator
2. Add `GET /users/me` API to get current user from jwt token

## Related Issue

https://github.com/lil-lab/recnet/issues/107

## Notes

### Auth Decorator usage
```ts
  @Get("me")
  @Auth()
  public async getMe(@User() authUser: AuthUser): Promise<GetUserMeResponse> {
    const { userId } = authUser;
    // call your service and return your response
  }
```

### How to test
1. point your local environment to dev postgres DB or clone dev DB to local DB
2. Find your user uuid in the database and hardcode in your `user.controller.ts`
```ts
public async getMe(@User() authUser: AuthUser): Promise<GetUserMeResponse> {
    let { userId } = authUser;
    userId = "your-user-uuid-id";
    const user = await this.userService.getUser(userId);
    return { user };
  }
```
This is a workaround because we have not finished the login API and migration. Thus, the userId in the JWT token comes from Firebase instead of the universal uuid in Postgres DB. It should be the user's uuid after the frontend fully migrates to our custom backend system.
3. run `nx serve recnet-api` on your localhost
4. Copy your Bearer authtoken from `recnet.io` current login API header
5. Use that token in header and hit `GET /users/me`. You can access the API either by postman or visit `localhost:3000/api` to execute on the swagger doc.

![Screenshot 2024-03-25 at 1 10 33 PM](https://github.com/lil-lab/recnet/assets/22555930/ae5abceb-8c2a-47c2-9657-e5ae26c9370e)



## TODO
